### ISSUE TITLE
gcloud build fails / Error when using _TEST_DATA

### ISSUE THREAD

**Issue by Faizaanahmed** [GoogleCloudPlatform/cortex-data-foundation#5]
Getting the below error even though all the buckets/BQ tables/gcloud build location is in EU.

```
starting build "3667fa7c-e603-4997-98b0-fe47c26c28c2"

FETCHSOURCE
Fetching storage object: gs://tsystems-cortex/digitribe-335411_cloudbuild/source/1650983390.276123-c95cd6b49d70426284a1b75fc263d40c.tgz#1650983391911618
Copying gs://tsystems-cortex/digitribe-335411_cloudbuild/source/1650983390.276123-c95cd6b49d70426284a1b75fc263d40c.tgz#1650983391911618...
/ [0 files][    0.0 B/  3.9 MiB]                                                
/ [1 files][  3.9 MiB/  3.9 MiB]                                                
Operation completed over 1 objects/3.9 MiB.                                      
BUILD
Pulling image: gcr.io/kittycorn-public/deploy-kittycorn:v1.2
v1.2: Pulling from kittycorn-public/deploy-kittycorn
f003217c5aae: Already exists
d1541c6ba716: Already exists
dfb3217166c8: Already exists
a1033864d5e1: Already exists
699b15d2c3b8: Already exists
6cb7a5182461: Pulling fs layer
026205fede29: Pulling fs layer
d8c5e66a9ac6: Pulling fs layer
3eaec9438278: Pulling fs layer
e5f65183c5dc: Pulling fs layer
7a7752f8dff9: Pulling fs layer
3eaec9438278: Waiting
e5f65183c5dc: Waiting
7a7752f8dff9: Waiting
026205fede29: Verifying Checksum
026205fede29: Download complete
6cb7a5182461: Download complete
d8c5e66a9ac6: Verifying Checksum
d8c5e66a9ac6: Download complete
3eaec9438278: Verifying Checksum
3eaec9438278: Download complete
7a7752f8dff9: Verifying Checksum
7a7752f8dff9: Download complete
6cb7a5182461: Pull complete
e5f65183c5dc: Verifying Checksum
e5f65183c5dc: Download complete
026205fede29: Pull complete
d8c5e66a9ac6: Pull complete
3eaec9438278: Pull complete
e5f65183c5dc: Pull complete
7a7752f8dff9: Pull complete
Digest: sha256:12e8ebb8320513fd5611824d11ae3c59f7687016a4d4a64bc34aa50db4bf6089
Status: Downloaded newer image for gcr.io/kittycorn-public/deploy-kittycorn:v1.2
gcr.io/kittycorn-public/deploy-kittycorn:v1.2
---Creating View: --- 
CREATE OR REPLACE FUNCTION `digitribe-335411.Tsystems_Cortex_REPORTING.Fiscal_Case3`(
  ip_mandt STRING, ip_periv STRING, ip_date DATE
) AS (
  (
    SELECT MIN(CONCAT(CAST(
          IF(Bdatj = '0000',
            CAST( EXTRACT(YEAR
                FROM
                Ip_Date)AS STRING),
            Bdatj) AS INT) + CAST(Reljr AS INT), "|", Poper))
    FROM
      `digitribe-335411.CDC_TABLES.t009b`
    WHERE
      Mandt = Ip_Mandt
      AND Periv = Ip_Periv
      AND CONCAT(
        IF(Bdatj = '0000',
          CAST( EXTRACT(YEAR
            FROM
            Ip_Date)AS STRING),
          Bdatj), Bumon, Butag) >= FORMAT_TIMESTAMP('%Y%m%d', Ip_Date) )
);
BigQuery error in query operation: Error processing job
'digitribe-335411:bqjob_re9b75f4ed0e8e38_000001806647855e_1': Not found: Table
digitribe-335411:CDC_TABLES.t009b was not found in location EU
ERROR
ERROR: build step 0 "gcr.io/kittycorn-public/deploy-kittycorn:v1.2" failed: step exited with non-zero status: 2
```
Seems like the table t009b is not being generated.

**Comment by Lsubatin** [comment]
Hi [Faizaanahmed](https://github.com/Faizaanahmed), did you receive an error in the CDC stage when creating that specific table? Are the other tables loaded in that dataset?

**Comment by Faizaanahmed** [comment]
@Lsubatin  yes, the error happens at the CDC stage. Only one table loaded -  csks_hier

**Comment by Lsubatin** [comment]
Then we need more details: have you executed the test harness (parameters _TEST_DATA=true)? if yes, is that dataset in region EU? can you paste the errors of the generation of the test harness (step 0 in the build it starts with "bash -c if [ 'true' = "true" ] ; then ./create_test_harness.sh")

**Comment by Faizaanahmed** [comment]
Yes, the _TEST_DATA=true command was used and the dataset is in EU region as well.

Command to start the deployment - 
`gcloud builds submit --project digitribe-335411 --gcs-source-staging-dir="gs://tsystems-cortex/digitribe-335411_cloudbuild/source" --substitutions  _PJID_SRC=digitribe-335411,_PJID_TGT=digitribe-335411,_DS_CDC=CDC_TABLES,_DS_RAW=1RAW_LANDING,_DS_REPORTING=Tsystems_Cortex_REPORTING,_DS_MODELS=Tsystems_Cortex_ML,_TGT_BUCKET=tsystems-cortex/digitribe-335411_Cortex_TGT,_TEST_DATA=true,_DEPLOY_CDC=false,_LOCATION=EU,_GCS_BUCKET=tsystems-cortex/digitribe-335411_Cortex_GSG`

The log for step 0:
`starting build "79cbbc51-c9d6-497e-986b-cec7a870ed32"

FETCHSOURCE
Fetching storage object: gs://tsystems-cortex/digitribe-335411_cloudbuild/source/1651034931.203317-62610bb80acc4287ae7e04b748e8ef8b.tgz#1651034933355900
Copying gs://tsystems-cortex/digitribe-335411_cloudbuild/source/1651034931.203317-62610bb80acc4287ae7e04b748e8ef8b.tgz#1651034933355900...
/ [0 files][    0.0 B/  3.9 MiB]                                                
/ [0 files][264.0 KiB/  3.9 MiB]                                                
-
- [1 files][  3.9 MiB/  3.9 MiB]                                                
Operation completed over 1 objects/3.9 MiB.                                      
BUILD
Starting Step #0
Step #0: Already have image (with digest): gcr.io/cloud-builders/gcloud
Step #0: Processing adr6
Step #0: Processing bkpf
Step #0: Processing ankt
Step #0: Processing anla
Step #0: Processing adrt
Step #0: Processing but000
Step #0: Processing bseg
Step #0: Processing adrct
Step #0: Processing adrc
Step #0: Processing but020
Step #0: Table adrt already exists, skipping
Step #0: Table adr6 already exists, skipping
Step #0: Table ankt already exists, skipping
Step #0: Table bseg already exists, skipping
Step #0: Processing cepc
Step #0: Table anla already exists, skipping
Step #0: Processing cepct
Step #0: Processing csks
Step #0: Processing dd03l
Step #0: Processing cskt
Step #0: Table adrc already exists, skipping
Step #0: Processing ekbe
Step #0: Table but000 already exists, skipping
Step #0: Processing ekes
Step #0: Table adrct already exists, skipping
Step #0: Table bkpf already exists, skipping
Step #0: Table but020 already exists, skipping
Step #0: Processing eket
Step #0: Processing ekko
Step #0: Processing ekkn
Step #0: Table cskt already exists, skipping
Step #0: Processing ekpo
Step #0: Table csks already exists, skipping
Step #0: Table cepct already exists, skipping
Step #0: Processing kna1
Step #0: Processing faglflexa
Step #0: Table cepc already exists, skipping
Step #0: Processing konv
Step #0: Table eket already exists, skipping
Step #0: Processing lfa1
Step #0: Table ekko already exists, skipping
Step #0: Processing likp
Step #0: Table ekbe already exists, skipping
Step #0: Table ekes already exists, skipping
Step #0: Processing lips
Step #0: Processing makt
Step #0: Table dd03l already exists, skipping
Step #0: Table ekkn already exists, skipping
Step #0: Processing mara
Step #0: Processing mard
Step #0: Table faglflexa already exists, skipping
Step #0: Table ekpo already exists, skipping
Step #0: Processing rbco
Step #0: Processing prps
Step #0: Table konv already exists, skipping
Step #0: Processing rbkp
Step #0: Table kna1 already exists, skipping
Step #0: Processing rseg
Step #0: Table makt already exists, skipping
Step #0: Processing setheadert
Step #0: Table mara already exists, skipping
Step #0: Processing setleaf
Step #0: Table lips already exists, skipping
Step #0: Processing setnode
Step #0: Table lfa1 already exists, skipping
Step #0: Processing ska1
Step #0: Table likp already exists, skipping
Step #0: Processing skat
Step #0: Table mard already exists, skipping
Step #0: Processing t001
Step #0: Table prps already exists, skipping
Step #0: Processing t001w
Step #0: Table rseg already exists, skipping
Step #0: Processing t002
Step #0: Table rbco already exists, skipping
Step #0: Processing t005
Step #0: Table setheadert already exists, skipping
Step #0: Table rbkp already exists, skipping
Step #0: Processing t005k
Step #0: Processing t005s
Step #0: Table skat already exists, skipping
Step #0: Processing t005t
Step #0: Table setnode already exists, skipping
Step #0: Processing t006
Step #0: Table ska1 already exists, skipping
Step #0: Processing t006a
Step #0: Table setleaf already exists, skipping
Step #0: Processing t006t
Step #0: Table t001 already exists, skipping
Step #0: Processing t009
Step #0: Table t001w already exists, skipping
Step #0: Processing t009b
Step #0: Table t005 already exists, skipping
Step #0: Processing t023
Step #0: Table t005s already exists, skipping
Step #0: Processing t023t
Step #0: Table t002 already exists, skipping
Step #0: Processing t134
Step #0: Table t005t already exists, skipping
Step #0: Processing t134t
Step #0: Table t005k already exists, skipping
Step #0: Processing t179
Step #0: Table t006a already exists, skipping
Step #0: Processing t179t
Step #0: Table t006 already exists, skipping
Step #0: Processing t881
Step #0: Table t006t already exists, skipping
Step #0: Table t009 already exists, skipping
Step #0: Processing t881t
Step #0: Processing tcurc
Step #0: Table t009b already exists, skipping
Step #0: Processing tcurf
Step #0: Table t023 already exists, skipping
Step #0: Processing tcurr
Step #0: Table t134 already exists, skipping
Step #0: Processing tcurt
Step #0: Table t023t already exists, skipping
Step #0: Processing tcurx
Step #0: Table t134t already exists, skipping
Step #0: Table t179 already exists, skipping
Step #0: Processing tka02
Step #0: Processing tvko
Step #0: Table t881t already exists, skipping
Step #0: Processing tvkot
Step #0: Table t881 already exists, skipping
Step #0: Processing tvtw
Step #0: Table t179t already exists, skipping
Step #0: Processing tvtwt
Step #0: Table tcurc already exists, skipping
Step #0: Processing vbak
Step #0: Table tcurf already exists, skipping
Step #0: Processing vbap
Step #0: Table tcurr already exists, skipping
Step #0: Processing vbep
Step #0: Table tka02 already exists, skipping
Step #0: Table tcurt already exists, skipping
Step #0: Processing vbfa
Step #0: Processing vbpa
Step #0: Table tcurx already exists, skipping
Step #0: Processing vbrk
Step #0: Table tvko already exists, skipping
Step #0: Processing vbrp
Step #0: Table tvtw already exists, skipping
Step #0: Processing vbuk
Step #0: Table tvkot already exists, skipping
Step #0: Processing vbup
Step #0: Table tvtwt already exists, skipping
Step #0: Table vbak already exists, skipping
Step #0: Table vbap already exists, skipping
Step #0: Table vbep already exists, skipping
Step #0: Table vbpa already exists, skipping
Step #0: Table vbfa already exists, skipping
Step #0: Table vbrp already exists, skipping
Step #0: Table vbuk already exists, skipping
Step #0: Table vbrk already exists, skipping
Step #0: Table vbup already exists, skipping
Finished Step #0`

This step was successful, althought I am getting an error on step 3.
![image](https://user-images.githubusercontent.com/25970667/165452879-bec42c5d-4165-4d1d-ab87-09d113df0d2f.png)

**Comment by Lsubatin** [comment]
Thanks for sending the commands and details, it's very helpful. I see you're using a CDC dataset but that dataset is not getting populated because _DEPLOY_CDC is set to false. For test data, the parameter should be true so the data is copied form the raw dataset into the CDC one.

**Comment by Faizaanahmed** [comment]
@fawix I cloned the repository again and tried with the modified command but still facing the same error as before. 
FYI -  I am adding the command the flag `--gcs-source-staging-dir=gs://tsystems-cortex/digitribe-335411_cloudbuild/source` in **deploy_cdc.sh** as my project is in the EU region and not setting it gives the ` ERROR: (gcloud.builds.submit) HTTPError 412: 'us' violates constraint 'constraints/gcp.resourceLocations'`error

**Comment by fawix** [comment]
Hi @Faizaanahmed, thank you! That helps with context. You seem to have an Org Policy enabled to restrict location, I'll test with that setup as well to troubleshoot the bucket contain issue.

For CDC:

Did the error remain even after you removed the prefix path from the `_TGT_BUCKET`? Could you add the logs for CDC and the screenshot for cloud build step details?

Thanks,
Fatima

**Comment by Faizaanahmed** [comment]
Hi @fawix , Thank you for the quick reply.
Yes the error still remains after removing the prefix path. Please find the details here : 
Logs :  [log-1bc2fae3-82d7-4a38-863b-35a067ffeba6.txt](https://github.com/GoogleCloudPlatform/cortex-data-foundation/files/8715577/log-1bc2fae3-82d7-4a38-863b-35a067ffeba6.txt)
Screenshot:
![image](https://user-images.githubusercontent.com/25970667/169018432-0551e68f-fdfd-4f3a-9366-cfe9359ead14.png)

**Comment by fawix** [comment]
Thanks @Faizaanahmed ! Can you also add a screenshot of the "execution details" tab? 

I'll try to use the same parameters to reproduce the error. 

Thanks 🙏

**Comment by Faizaanahmed** [comment]
@fawix Sure

![image](https://user-images.githubusercontent.com/25970667/169024827-c05503ed-5f0e-4ef6-a3e6-8e150ac2b6d2.png)

Thanks

**Comment by benschuler** [comment]
Hi @Faizaanahmed, I was able to reproduce the issue you are describing here with similar parameters and the same error message saying `Missing whitespace between literal and alias`. After some deeper investigation I found out that this is likely caused by the name of the dataset `1RAW_LANDING` starting with a number. When I picked this name for the dataset, the deployment failed at the same stage as your logs suggested, but when I change only this parameter to `RAW_LANDING` without making any other changes the deployment executes without errors.

Can you please check whether this is a feasible workaround for you at this time and your deployment executes successfully?

**Comment by gvrvpndt** [comment]
This issue is fixed in release 2.2, and it should now work even when a dataset name is starting with a number.

For release prior to 2.2, @benschuler's workaround should help. For Release 2.2, you should not need any workaround.

**Comment by Faizaanahmed** [comment]
Thank you @benschuler @gvrvpndt I was able to deploy the tables with the workaround.
